core/src/test/java/org/apache/iceberg/rest/responses/TestUpdateNamespacePropertiesResponse.java

```diff
@@ -0,0 +1,174 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.responses;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.util.JsonUtil;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+public class TestUpdateNamespacePropertiesResponse {
+
+  private static final ObjectMapper MAPPER = JsonUtil.withRESTComponents();
+
+  private static final String[] ALL_FIELDS_FROM_SPEC = { "updated", "removed", "missing" };
+
+  /* Values used to fill in response fields */
+  private static final List<String> UPDATED = ImmutableList.of("owner");
+  private static final List<String> REMOVED = ImmutableList.of("foo");
+  private static final List<String> MISSING = ImmutableList.of("bar");
+
+  @Test
+  public void testValidSerDe() throws JsonProcessingException {
+    Map<String, UpdateNamespacePropertiesResponse> toTest = ImmutableMap.of(
+        // Full response
+        "{\"removed\":[\"foo\"],\"updated\":[\"owner\"],\"missing\":[\"bar\"]}",
+        builder().addUpdated(UPDATED).addRemoved(REMOVED).addMissing(MISSING).build(),
+        // Only updated
+        "{\"removed\":null,\"updated\":[\"owner\"],\"missing\":null}",
+        builder().addUpdated(UPDATED).build(),
+        "{\"removed\":[],\"updated\":[\"owner\"],\"missing\":[]}",
+        builder().addUpdated(UPDATED).addMissing(ImmutableList.of()).addRemoved(ImmutableList.of()).build(),
+        // Only removed
+        "{\"removed\":[\"foo\"],\"updated\":null,\"missing\":null}",
+        builder().addRemoved(REMOVED).build(),
+        "{\"removed\":[\"foo\"],\"updated\":[],\"missing\":[]}",
+        builder().addRemoved(REMOVED).addMissing(ImmutableList.of()).addUpdated(ImmutableList.of()).build(),
+        // Only missing
+        "{\"removed\":null,\"updated\":null,\"missing\":[\"bar\"]}",
+        builder().addMissing(MISSING).build(),
+        "{\"removed\":[],\"updated\":[],\"missing\":[\"bar\"]}",
+        builder().addMissing(MISSING).addUpdated(ImmutableList.of()).addRemoved(ImmutableList.of()).build());
```

Comment:
This is the form I'm planning on using for the valid SerDe tests so the named constants for the JSON can be inlined.